packages/cardano-services/src/Program/programs/providerServer.ts

```diff
@@ -222,12 +223,33 @@ const serviceMapFactory = (options: ServiceMapFactoryOptions) => {
       });
       return new NetworkInfoHttpService({ logger, networkInfoProvider });
     }, ServiceNames.NetworkInfo),
-    [ServiceNames.TxSubmit]: async () => {
+    [ServiceNames.TxSubmit]: withDbSyncProvider(async (dbPools, cardanoNode) => {
+      if (!genesisData)
+        throw new MissingProgramOption(
+          ServiceNames.NetworkInfo,
+          ProviderServerOptionDescriptions.CardanoNodeConfigPath
+        );
+      const networkInfoProvider = new DbSyncNetworkInfoProvider({
```

Comment:
@mkazlauskas I think this is the right thing to do, for set up requirements, but it might not be.

Reply:
A single instance of DbSyncNetworkInfoProvider should be reused for both:
- here as a dependency for kora labs provider
- [ServiceNames.NetworkInfo]

It should be created lazily, so that it's not instantiated when neither `ServiceNames.TxSubmit` nor `ServiceNames.NetworkInfo` is enabled

Reply:
9a0c19b5dc62fd220c9d1414e8cc59b646a50a97 should fix this